packages/recommend-react/src/__tests__/Recommend.test.tsx

```diff
@@ -0,0 +1,110 @@
+import { act, render, waitFor } from '@testing-library/react';
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import { createMultiSearchResponse } from '../../../../test/utils';
+import {
+  createRecommendClient,
+  hit,
+} from '../../../../test/utils/createRecommendClient';
+import { Recommend, useRecommendContext } from '../Recommend';
+
+function createMockedRecommendClient() {
+  const recommendClient = createRecommendClient({
+    getRecommendations: jest.fn(() =>
+      Promise.resolve(
+        createMultiSearchResponse({
+          hits: [hit, hit],
+        })
+      )
+    ),
+  });
+
+  return {
+    recommendClient,
+  };
+}
+
+describe('Recommend', () => {
+  const onResultSpy = jest.fn();
+
+  const RecommendComponent = ({
+    objectId,
+    model,
+  }: {
+    objectId: string;
+    model: string;
+  }) => {
+    const { register, hasProvider } = useRecommendContext();
+
+    React.useEffect(() => {
+      const key = JSON.stringify({
+        k: objectId,
+      });
+      register({
+        key,
+        getParameters: () => {
+          return {
+            keyPair: {
+              key,
+              value: 1,
+            },
+            queries: [
+              {
+                indexName: 'index',
+                model,
+                objectID: objectId,
+                maxRecommendations: 10,
+              },
+            ],
+          };
+        },
+        onRequest: () => {},
+        onResult: (result) => {
+          onResultSpy(result);
+        },
+      });
+    }, [register, objectId, model]);
+    return (
+      <div>
+        <div>hasProvider:{`${hasProvider}`}</div>
+      </div>
+    );
+  };
```

Comment:
The public API is exposed through the hooks. So that has already been tested. 

Since we don't expose internal logic this was my attempt at capturing and testing the life-cycle of the provider. I can't think of another way.  

But indeed this is is already tested through the individual hooks.